activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -48,11 +48,11 @@ def writer(records)
       # Implements the ids reader method, e.g. foo.item_ids for Foo.has_many :items
       def ids_reader
         if loaded?
-          target.pluck(reflection.association_primary_key)
+          target.pluck(*reflection.association_primary_key)
         elsif !target.empty?
-          load_target.pluck(reflection.association_primary_key)
+          load_target.pluck(*reflection.association_primary_key)
         else
-          @association_ids ||= scope.pluck(reflection.association_primary_key)
+          @association_ids ||= scope.pluck(*reflection.association_primary_key)
```

Comment:
The doc show pluck with multiple arguments like `(:id, :name)`, but `([:id, :name])` also works. To align with docs, we should use add the splat operator here. 

https://github.com/rails/rails/blob/23af6f37a310e16916ca788837f4a7accb2cebad/activerecord/lib/active_record/relation/calculations.rb#L266